From: Michael Olbrich <m.olbrich@pengutronix.de>
Subject: fix zlib handling for host tools

The host-tools build system is broken. It uses the zlib switch for the
target to determine if the system or internal zlib should be used.
This breaks if we select system zlib for the target and there is no zlib.h
installed on the build system.
It is also not possible to add aditional include paths for host tools.

With this patch the host-tools always use the internal zlib when
cross-compiling.
Fixed upstream for Qt 4.7

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>

---
 src/tools/bootstrap/bootstrap.pri |    2 +-
 src/tools/bootstrap/bootstrap.pro |    2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

Index: b/src/tools/bootstrap/bootstrap.pro
===================================================================
--- a/src/tools/bootstrap/bootstrap.pro
+++ b/src/tools/bootstrap/bootstrap.pro
@@ -95,7 +95,7 @@ macx: {
    LIBS += -framework CoreServices
 }
 
-contains(QT_CONFIG, zlib) {
+contains(QT_CONFIG, zlib)|cross_compile {
    INCLUDEPATH += ../../3rdparty/zlib
    SOURCES+= \
         ../3rdparty/zlib/adler32.c \
Index: b/src/tools/bootstrap/bootstrap.pri
===================================================================
--- a/src/tools/bootstrap/bootstrap.pri
+++ b/src/tools/bootstrap/bootstrap.pri
@@ -51,7 +51,7 @@ hpux-acc*|hpuxi-acc* {
     }
     LIBS += -lbootstrap
 }
-!contains(QT_CONFIG, zlib):!contains(QT_CONFIG, no-zlib) {
+!contains(QT_CONFIG, zlib):!contains(QT_CONFIG, no-zlib):!cross_compile {
    unix:LIBS += -lz
 #  win32:LIBS += libz.lib
 }
